identity_presentation/src/commonMain/kotlin/net/thechance/mena/identity/presentation/screen/pickLocation/PickLocationScreen.kt

```diff
@@ -0,0 +1,121 @@
+import org.jetbrains.compose.resources.stringResource
+import org.koin.core.parameter.parametersOf
+
+data class PickLocationScreen(
+    private val addressModel: AddressModel?,
+    private val onUpdateLocation: (AddressModel) -> Unit,
+) : BaseScreen<PickLocationScreenViewModel,
+        PickLocationScreenUIState,
+        PickLocationScreenUIEffect,
+        PickLocationScreenInteractionListener>() {
+
+    @Composable
+    override fun Content() {
+        InitScreen(getScreenModel(parameters = { parametersOf(addressModel) }))
+    }
+
+    @Composable
+    override fun OnRender(
+        state: PickLocationScreenUIState,
+        listener: PickLocationScreenInteractionListener
+    ) {
+        Scaffold(
+            topBar = {
+                AuthAppBar(
+                    title = stringResource(Res.string.pick_location_title),
+                    onClickBack = listener::onClickBack,
+                    backgroundColor = Theme.colorScheme.background.surfaceLow
+                )
+            }
+        ) {
+            PickLocationMap(
+                cameraPosition = state.cameraPosition,
+                onCameraMoved = listener::onMoveCamera,
+                onMapClick = listener::onClickMap,
+                anchorLocation = state.pointerLocation,
+                isLocked = state.isMapLocked,
+                currentLocation = state.currentLocation,
+                animateToCurrentLocation = state.animateToCurrentLocation,
+                onSetAnchorLocation = listener::onSetAnchorLocation
+            ) {
+                Column(
+                    Modifier.padding(Theme.spacing._16).fillMaxSize(),
+                    verticalArrangement = Arrangement.Bottom
+                ) {
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+                        GpsFabButton(
+                            onClick = listener::onClickGps,
+                            isLoading = state.isGpsButtonLoading,
+                            modifier = Modifier.padding(bottom = Theme.spacing._12)
+                        )
+                        EditMapButton(
+                            isMapLocked = state.isMapLocked,
+                            onEditClick = listener::onClickEdit
+                        )
+                    }
+                    PrimaryButton(
+                        text = stringResource(Res.string.confirm),
+                        onClick = listener::onClickConfirm,
+                        isEnabled = state.isConfirmEnabled,
+                        isLoading = state.isLoading,
+                        contentPadding = PaddingValues(vertical = 13.dp),
+                        modifier = Modifier.fillMaxWidth()
+                    )
+                }
+
+            }
+        }
+        ErrorSnackBar(
```

Comment:
You can move this to the `Scaffold` snackBar attribute
```suggestion
Scaffold(
      snackBar = {
          ErrorSnackBar(
                errorMessage = state.errorMessage,
                onDismiss = listener::onClearErrorMessage,
                modifier = Modifier.statusBarsPadding()
          )
```